src/Microsoft.VisualStudio.Services.Agent/HostContext.cs

```diff
@@ -175,6 +169,65 @@ public HostContext(HostType hostType, string logFile = null)
             }
         }
 
+        private ILoggedSecretMasker CreateSecretMasker()
+        {
+            // When enabled, use the OSS package-provided secret masker from
+            // Microsoft.Security.Utilities.Core. Otherwise, use the legacy
+            // secret masker from VSO.
+            bool useNewSecretMasker = AgentKnobs.EnableNewSecretMasker.GetValue(this).AsBoolean();
+
+            // When enabled, use add additional regex patterns to the secret
+            // masker. For the OSS package-provided secret masker, this will use
+            // its 'PreciselyClassifiedSecurityKeys'. For the legacy secret
+            // masker, this will use 'CredScanPatterns' implemented in this
```

Comment:
>CredScanPatterns [](http://example.com/codeflow?start=38&length=16)

This name bothers me, it really miscommunicates today's reality. It's more work, but we should consider doing some additional clean-up, e.g., deleting the `tools\CredScanRegexes` content and generally eliminating this name from the code base. 

That means renaming `AdditionalMaskingRegexes.CredScan.cs` and the public field it declares. In fact, all these regexes are `SecurityUtilitiesRegexes`.

Reply:
One option that I discussed elsewhere is to have one single flag that controls both new masker and additional regexes. Then we should simply delete the regexes from this repo.

I will wait to see what we decide on that before renaming anything.

Reply:
since we're keeping both options at least for now, I doing as you suggest.